### PR TITLE
fix(JobCard): onBookmarkClick not working

### PR DESCRIPTION
**Pull request by cm-seekasia**
Now, onBookmarkClick is not working. Need to fix it. 

Trello:
https://trello.com/c/QqJIpZHh/437-bug-saved-job-not-working-on-srp

### CODE REVIEW

File: react/JobCard/JobCard.demo.js [hunk 1]
```diff
@@ -295,16 +295,14 @@ export default {
           label: 'Bookmarked',
           transformProps: props => ({
             ...props,
-            bookmarked: 'bookmarked',
-            onBookmarkClick: () => {}
+            bookmarked: 'bookmarked'
```
Comment (swetharajendra05) [inline react/JobCard/JobCard.demo.js]: can we reuse the constant for `bookmarked` or `notbookmarked`. I see that being used on the JobCard.js.

File: react/JobCard/__snapshots__/JobCard.test.js.snap [hunk 2]
```diff
@@ -317,10 +317,9 @@ exports[`JobCard should pass through bookmark props 1`] = `
 >
   <Button
     aria-pressed={false}
-    bookmarked="bookmarked"
```
Comment (kosanna) [inline react/JobCard/__snapshots__/JobCard.test.js.snap]: That doesn't feel right

Comment (kosanna) [reply react/JobCard/__snapshots__/JobCard.test.js.snap]: Ah hang on, that's a button 🤔

**Comment by cm-seekasia** [comment]
> Once demo has variables good to go. I would love to see the test there as well for that so we don't break it again. But happy if it's done as a separate PR.

We do have test case for that => https://github.com/seekinternational/seek-asia-style-guide/blob/d06f5e140546c280876bba5bcdb91c21f4a2675d/react/JobCard/__snapshots__/JobCard.test.js.snap#L322
But we missed to check.

**Pull request merged**